Client/Frontend/Library/RecentlyClosedTabsPanel.swift

```diff
@@ -118,9 +118,21 @@ class RecentlyClosedTabsPanelSiteTableViewController: SiteTableViewController {
 
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
         tableView.deselectRow(at: indexPath, animated: true)
-        recentlyClosedTabsDelegate?.openRecentlyClosedSiteInNewTab(recentlyClosedTabs[indexPath.row].url, isPrivate: false)
-        let visitType = VisitType.typed    // Means History, too.
-        libraryPanelDelegate?.libraryPanel(didSelectURL: recentlyClosedTabs[indexPath.row].url, visitType: visitType)
+        let url = recentlyClosedTabs[indexPath.row].url
+        recentlyClosedTabsDelegate?.openRecentlyClosedSiteInNewTab(url, isPrivate: false)
+
+        // The code above creates new tab and selects it, but TabManagerImplementation.selectTab()
+        // currently performs the actual selection update asynchronously via Swift Async + Task/Await.
+        // This means that selectTab() returns before the tab is actually selected. As a result, the
+        // delegate callback below will incorrectly cause a duplicate tab (since it will treat the
+        // url as having been applied to the current tab, not our newly-added tab from above). This
+        // is avoided by making sure we wait for our expected tab above to be selected before
+        // notifying our library panel delegate. [FXIOS-7741]
+
+        AppEventQueue.wait(for: .selectTab(url)) {
+            let visitType = VisitType.typed    // Means History, too.
```

Comment:
I'm not sure what `// Means History, too.` means?

Reply:
This was part of the previous code, I assumed it was just to clarify that the `.typed` VisitType was correct for a URL opened from the History panel, but I'm not 100% sure on the context.

Reply:
Ah the `VisitType` comes from Rust places. Let's leave it be then